src/Compilers/CSharp/Portable/Symbols/SubstitutedMethodSymbol.cs

```diff
@@ -424,23 +435,12 @@ public sealed override bool Equals(Symbol obj, TypeCompareKind compareKind)
 
         public override int GetHashCode()
         {
-            int code = _hashCode;
-
-            if (code == 0)
+            if (!_hashCode.HasValue)
             {
-                code = ComputeHashCode();
-
-                // 0 means that hashcode is not initialized. 
-                // in a case we really get 0 for the hashcode, tweak it by +1
-                if (code == 0)
-                {
-                    code++;
-                }
-
-                _hashCode = code;
+                _hashCode = ComputeHashCode();
```

Comment:
>_hashCode = ComputeHashCode(); [](start = 16, length = 30)

This assignment is no longer atomic. I recommend reverting the type change and moving increment into the ComputeHashCode.